crates/cache/src/oci.rs

```diff
@@ -0,0 +1,130 @@
+//! Cache for OCI registry entities.
+
+use anyhow::{Context, Result};
+use oci_distribution::Reference;
+use tokio::fs;
+
+use std::path::{Path, PathBuf};
+
+const CONFIG_DIR: &str = "fermyon";
+const REGISTRY_CACHE_DIR: &str = "registry";
+const OCI_CACHE_DIR: &str = "oci";
+const MANIFESTS_DIR: &str = "manifests";
+const WASM_DIR: &str = "wasm";
+const DATA_DIR: &str = "data";
+
+/// Cache for registry entities.
+pub struct Cache {
+    /// Root directory for the cache instance.
+    pub root: PathBuf,
+}
+
+impl Cache {
+    /// Create a new cache given an optional root directory.
+    pub async fn new(root: Option<PathBuf>) -> Result<Self> {
+        let root = match root {
+            Some(root) => root,
+            None => dirs::config_dir()
+                .context("cannot get configuration directory")?
+                .join(CONFIG_DIR),
+        };
+        let root = root.join(REGISTRY_CACHE_DIR).join(OCI_CACHE_DIR);
+        Self::ensure_dirs(&root).await?;
+
+        Ok(Self { root })
+    }
+
+    /// The manifests directory for the current cache.
+    pub fn manifests_dir(&self) -> PathBuf {
```

Comment:
Is it desirable that these are `pub`, or should we aspire to make the structure the concern of the cache, and hand out specific file paths or content on demand, rather than expecting consumers to construct file paths into the cache?

Reply:
Updated these to be private and added functions to directly return the file path based on digest.